caim_base/utils/salesforce.py

```diff
@@ -0,0 +1,69 @@
+from django.conf import settings
+from simple_salesforce import Salesforce
+
+
+def _user_from_form(user_form):
+    '''compile user data from form into object for salesforce'''
+    form_data = user_form.cleaned_data
+
+    return {
+        "FirstName": form_data.get("first_name"),
+        "LastName": form_data.get("last_name"),
+        "Email": form_data.get("email"),
+        "MailingCity": form_data.get("city"),
+        "MailingState": form_data.get("state"),
+        "MailingPostalCode": form_data.get("zip_code"),
+    }
+
+
+def _salesforce_connection():
+    '''create salesforce connection with auth'''
+    sf_username = settings.SALESFORCE_USERNAME
+    sf_password = settings.SALESFORCE_PASSWORD
+    sf_security_token = settings.SALESFORCE_SECURITY_TOKEN
+
+    return Salesforce(
```

Comment:
should this be implemented as a singleton instead? Is the `Salesforce` class already doing that for us? My concern is if this is sending some kind of login API request every time this class is instantiated.

Reply:
okay I followed some of the logic in the source code of this library and it does seem that the `Salesforce` class issues a login API call every time this class is instantiated

https://github.com/simple-salesforce/simple-salesforce/blob/e3393bbcaf3d9651f11e3014c3c8cdc44d39a305/simple_salesforce/api.py#L30

Reply:
The Salesforce login request is instantiated as needed (on registration or profile update) as the function creating the request is only called then. It is destroyed when the function(s) calling to `salesforce.py` end.

We wouldn't try to keep an open connection I don't believe by opening the connection on app run or something (and I sort of doubt this is possible anyway as Salesforce likely would terminate a long running connection or it would go stale). So this opens the connection to Salesforce when needed, does the appropriate work, and then the connection dies as it goes out of scope at termination of the calling function. 

I believe this is how it should be given the required operations. (It might be different if we had multiple operations happening at a faster pace. In which case it would make sense to try to re-use connections).